src/app/data-client.ts

```diff
@@ -53,11 +53,21 @@ export class DataClient {
         break;
       }
       dataArray.push(
-        ...dataList.map((data) => ({
-          ...data.toObject(),
-          timeRequested: data.getTimeRequested()?.toDate(),
-          timeReceived: data.getTimeReceived()?.toDate(),
-        }))
+        ...dataList.map((data) => {
+          const mdIndex = data.getMetadataIndex();
+          const mdListLength = response.getMetadataList().length;
+          if (mdListLength != 0 && mdIndex >= mdListLength) {
+            throw Error(
```

Comment:
I wouldn't error, would just have metadata be blank instead (and document that metadata might be blank in the method docstring) - the data is likely still useful in that case